okhttp/src/test/java/okhttp3/ConscryptTest.java

```diff
@@ -31,23 +32,22 @@
 import static org.junit.Assert.assertTrue;
 
 public class ConscryptTest {
-  private OkHttpClient client = new OkHttpClient();
-
   @BeforeClass
-  public static void setup() {
-    assumeConscrypt();
-
-    OpenSSLProvider provider = new OpenSSLProvider();
-    Security.insertProviderAt(provider, 1);
+  public static void installProvider() {
+    if ("conscrypt".equals(System.getProperty("okhttp.platform"))) {
```

Comment:
This is the main fix.  We need to insert the provider before the JDK and OkHttp cache anything SSL related.

But it also means that assumptions in ConnectionSpecTest break.  And some behaviour differs in Intellij running one test vs running in gradle with daemon or locally.  Hence the bad commit.

Reply:
Tests just passed.... but....


```
okhttp3.ConscryptTest > testMozilla SKIPPED

okhttp3.ConscryptTest > testGoogle SKIPPED
```

Reply:
Want a method like `Platform.reset()` ?

Reply:
I want internal methods to reset to null or a specific Platform instance :)

But generally if we have only a few places in the code that call Platform.get() e.g. an internal var of OkHttp client it makes this testing soooooo much easier.

Reply:
Specifically Platform.reset() is still a JVM singleton, which makes strong assumptions about how the tests run in parallel.  It's not clear to me if gradle parallel is within same test runner instance or not.